spec/system/nested_imageable_spec.rb

```diff
@@ -0,0 +1,264 @@
+      fill_required_fields_for factory, path
+
+      imageable_attach_new_file(file_fixture("clippy.jpg"))
+
+      click_button submit_button_text
+
+      expect(page).to have_content notice_text
+    end
+
+    scenario "Should show new image after successful creation with one uploaded file" do
+      fill_required_fields_for factory, path
+
+      imageable_attach_new_file(file_fixture("clippy.jpg"))
+
+      click_button submit_button_text
+
+      expect(page).to have_content notice_text
+
+      if factory == :budget
+        click_link "Go back to edit budget"
+      else
+        expect(page).to have_css "figure figcaption"
+      end
+      expect(page).to have_css "figure img"
+    end
+
+    scenario "Different URLs for different images" do
+      imageable_attach_new_file(file_fixture("clippy.jpg"))
+
+      original_src = find(:fieldset, "Descriptive image").find("img")[:src]
+
+      click_link "Remove image"
+      imageable_attach_new_file(file_fixture("custom_map.jpg"))
+
+      updated_src = find(:fieldset, "Descriptive image").find("img")[:src]
+
+      expect(updated_src).not_to eq original_src
+    end
+  end
+
+  context "Only for edit path" do
+    let(:factory) { :proposal }
+    let(:path) { edit_proposal_path(imageable) }
+
+    before do
+      create(:image, imageable: imageable)
+      imageable.update!(author: user)
+      login_as(user)
+      visit path
+    end
+
+    scenario "show persisted image" do
+      expect(page).to have_css ".image-fields", count: 1
+      expect(page).not_to have_css "a#new_image_link"
+    end
+
+    scenario "remove nested field after removing the image" do
+      click_link "Remove image"
+
+      expect(page).not_to have_css ".image-fields"
+      expect(page).to have_link id: "new_image_link"
+    end
+
+    scenario "don't duplicate fields after removing and adding an image" do
+      click_link "Remove image"
+      click_link "Add image"
+
+      expect(page).to have_css ".image-fields", count: 1, visible: :all
+    end
+  end
+
+  def fill_required_fields_for(factory, path)
+    return if edit_path?(path)
+
+    if factory == :budget
+      fill_budget
+    elsif factory == :budget_investment
+      fill_budget_investment
+    elsif factory == :proposal
+      fill_proposal
+    end
+  end
+
+  def fill_proposal
+    fill_in_new_proposal_title with: "Proposal title"
+    fill_in "Proposal summary", with: "Proposal summary"
+    check :proposal_terms_of_service
+  end
+
+  def fill_budget
+    fill_in "Name", with: "Budget name"
+  end
+
+  def fill_budget_investment
+    fill_in_new_investment_title with: "Budget investment title"
+    fill_in_ckeditor "Description", with: "Budget investment description"
+    check :budget_investment_terms_of_service
+  end
+
+  def admin_section?(path)
+    path.starts_with?("/admin/")
+  end
+
+  def management_section?(path)
+    path.starts_with?("/management/")
+  end
+
+  def edit_path?(path)
+    path.match?("/edit")
```

Comment:
```suggestion
    path.match?("/edit/")
```

So something like `/editors` doesn't match.

Reply:
Sorry, I meant:

```suggestion
    path.end_with?("/edit")
```